packages/serverless-logic-web-tools/src/homepage/recentModels/WorkspacesTableRow.tsx

```diff
@@ -0,0 +1,209 @@
+
+export type WorkspacesTableRowProps = {
+  rowIndex: TdSelectType["rowIndex"];
+  rowData: WorkspacesTableRowData;
+  isSelected: boolean;
+  /**
+   * event fired when the Checkbox is toggled
+   */
+  onToggle: (selected: boolean) => void;
+};
+
+export function WorkspacesTableRow(props: WorkspacesTableRowProps) {
+  const { isSelected, rowIndex } = props;
+  const { descriptor, editableFiles, totalFiles, name, isWsFolder, workspaceId, createdDateISO, lastUpdatedDateISO } =
+    props.rowData;
+  const workspaces = useWorkspaces();
+
+  const linkTo = useMemo(
+    () =>
+      routes.workspaceWithFilePath.path({
+        workspaceId: editableFiles[0].workspaceId,
+        fileRelativePath: editableFiles[0].relativePathWithoutExtension,
+        extension: editableFiles[0].extension,
+      }),
+    [editableFiles]
+  );
+
+  return (
+    <Tr key={name}>
+      <Td
+        select={{
+          rowIndex,
+          onSelect: (_event, checked) => props.onToggle(checked),
+          isSelected,
+        }}
+      />
+      <Td dataLabel={columnNames.name}>
+        {isWsFolder ? (
+          <>
+            <FolderIcon />
+            &nbsp;&nbsp;&nbsp;{name}
+          </>
+        ) : (
+          <>
+            <TaskIcon />
+            &nbsp;&nbsp;&nbsp;<Link to={linkTo}>{name}</Link>
+          </>
+        )}
+      </Td>
+      <Td dataLabel={columnNames.type}>
+        {isWsFolder ? <WorkspaceLabel descriptor={descriptor} /> : <FileLabel extension={editableFiles[0].extension} />}
+      </Td>
+      <Td dataLabel={columnNames.created}>
+        <RelativeDate date={new Date(createdDateISO ?? "")} />
+      </Td>
+      <Td dataLabel={columnNames.lastUpdated}>
+        <RelativeDate date={new Date(lastUpdatedDateISO ?? "")} />
+      </Td>
+      <Td dataLabel={columnNames.editableFiles}>{editableFiles.length}</Td>
+      <Td dataLabel={columnNames.totalFiles}>{totalFiles}</Td>
+      <Td isActionCell>
+        <ActionsColumn
+          items={[
+            {
+              title: "Delete",
+              onClick: () => workspaces.deleteWorkspace({ workspaceId }),
+            },
+          ]}
+        />
+      </Td>
+    </Tr>
+  );
+}
+
+export function WorkspacesTableRowError(props: { rowData: WorkspacesTableRowData }) {
+  const { rowData } = props;
+  const workspaces = useWorkspaces();
+
+  return (
+    <>
+      <Tr>
+        <Td>&nbsp;</Td>
+        <Td colSpan={Object.keys(columnNames).length}>
+          <ExclamationTriangleIcon />
+          &nbsp;&nbsp;
+          {workspacesTableRowErrorContent}&nbsp;
+          <Popover
+            maxWidth="30%"
+            bodyContent={
+              <>
+                Error obtaining information for the following element:
+                <br />
+                workspace name: <b>${rowData.descriptor.name}</b>
+                <br />
+                workspace id: <b>${rowData.workspaceId}</b>
```

Comment:
```suggestion
                workspace name: <b>{rowData.descriptor.name}</b>
                <br />
                workspace id: <b>{rowData.workspaceId}</b>
```
The `$` is being presented in the UI.